src/backend/Shared/EventBus/EventBus.Abstraction/InMemoryEventBusSubscriptionsManager.cs

```diff
@@ -0,0 +1,166 @@
+    {
+        // FIELDS
+        private readonly IDictionary<string, List<SubscriptionInfo>> _handlers;
+        private readonly ICollection<Type> _eventTypes;
+
+        // EVENTS
+        public event EventHandler<string> OnEventRemoved;
+
+        // CONSTRUCTORS
+        public InMemoryEventBusSubscriptionsManager()
+        {
+            _handlers = new Dictionary<string, List<SubscriptionInfo>>();
+            _eventTypes = new List<Type>();
+        }
+
+        // PROPERTIES
+        public bool IsEmpty => !_handlers.Keys.Any();
+
+        // METHODS
+        #region Add Subscription
+        public void AddDynamicSubscription<TH>(string eventName)
+            where TH : IDynamicIntegrationEventHandler
+        {
+            DoAddSubscription(typeof(TH), eventName, isDynamic: true);
+        }
+
+        public void AddSubscription<T, TH>()
+            where T : IntegrationEvent
+            where TH : IIntegrationEventHandler<T>
+        {
+            string eventName = GetEventKey<T>();
+
+            DoAddSubscription(typeof(TH), eventName, isDynamic: false);
+
+            if (!_eventTypes.Contains(typeof(T)))
+            {
+                _eventTypes.Add(typeof(T));
+            }
+        }
+
+        private void DoAddSubscription(Type handlerType, string eventName, bool isDynamic)
+        {
+            if (!HasSubscriptionsForEvent(eventName))
+            {
+                _handlers.Add(eventName, new List<SubscriptionInfo>());
+            }
+
+            if (_handlers[eventName].Any(s => s.HandlerType == handlerType))
+            {
+                throw new ArgumentException($"Handler Type {handlerType.Name} already registered for '{eventName}'", nameof(handlerType));
+            }
+
+            if (isDynamic)
+            {
+                _handlers[eventName].Add(SubscriptionInfo.Dynamic(handlerType));
+            }
+            else
+            {
+                _handlers[eventName].Add(SubscriptionInfo.Typed(handlerType));
+            }
+        }
+        #endregion
+
+        #region Remove Subscription
+        public void Clear() => _handlers.Clear();
+
+        public void RemoveDynamicSubscription<TH>(string eventName)
+            where TH : IDynamicIntegrationEventHandler
+        {
+            SubscriptionInfo handlerToRemove = FindDynamicSubscriptionToRemove<TH>(eventName);
+            DoRemoveHandler(eventName, handlerToRemove);
+        }
+
+        public void RemoveSubscription<T, TH>()
+            where TH : IIntegrationEventHandler<T>
+            where T : IntegrationEvent
+        {
+            SubscriptionInfo handlerToRemove = FindSubscriptionToRemove<T, TH>();
+            string eventName = GetEventKey<T>();
+            DoRemoveHandler(eventName, handlerToRemove);
+        }
+
+        private void DoRemoveHandler(string eventName, SubscriptionInfo subsToRemove)
+        {
+            if (subsToRemove == null) return;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/iamprovidence/Lama/pullRequest?prid=4794247)